distribution/xfer/download.go

```diff
@@ -35,9 +39,10 @@ func (ldm *LayerDownloadManager) SetConcurrency(concurrency int) {
 // NewLayerDownloadManager returns a new LayerDownloadManager.
 func NewLayerDownloadManager(layerStore layer.Store, concurrencyLimit int, options ...func(*LayerDownloadManager)) *LayerDownloadManager {
 	manager := LayerDownloadManager{
-		layerStore:   layerStore,
-		tm:           NewTransferManager(concurrencyLimit),
-		waitDuration: time.Second,
+		layerStore:      layerStore,
+		tm:              NewTransferManager(concurrencyLimit),
+		waitDuration:    time.Second,
+		cacheDirManager: cachedir.NewManager(filepath.Join(os.TempDir(), "resumable_downloads")),
```

Comment:
I don't think we want to store these temporary files underneath `os.TempDir()` for a few reasons:

- They will be harder for a user to locate, since they would be in a different place from other files created by `dockerd`.
- `/tmp` can sometimes be a memory filesystem or a small partition, which wouldn't be appropriate for long-term storage.
- `/tmp` is often reaped on a regular schedule, with inactive files deleted automatically.
- `/tmp` is world-readable, so a malicious user could potentially do something like create a symlink to `/etc/passwd` at a path we'd be expected to write to.

I think we should put this under the daemon's `config.Root` directory instead.

As for the name of the directory, `"partial_downloads"`? `"incomplete_downloads"`?